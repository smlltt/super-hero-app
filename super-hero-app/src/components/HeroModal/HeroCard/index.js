import React from "react";
import {
  Card,
  CardHeader,
  Divider,
  List,
  ListItemText,
} from "@material-ui/core";
import {
  CardContentWrapper,
  CardTextWrapper,
  HeaderWrapper,
  HeroCardMedia,
  StatBioTypography,
} from "../styles";
import IconButton from "@material-ui/core/IconButton";
import CloseIcon from "@material-ui/icons/Close";

const HeroCard = ({ selectedHero, closeSelectedHero }) => {
  const stats = [
    "intelligence",
    "strength",
    "speed",
    "durability",
    "power",
    "combat",
  ];

  const bioFacts = [
    "full-name",
    "alter-egos",
    "aliases",
    "place-of-birth",
    "first-appearance",
    "publisher",
    "alignment",
  ];

  const bioFactsList = bioFacts.map((bioFact) => {
    let capitalizedBioFact = bioFact[0].toUpperCase() + bioFact.slice(1);
    capitalizedBioFact = capitalizedBioFact.replace(/-/g, " ");
    return (
      <ListItemText>
        <StatBioTypography variant={"p"}>
          {capitalizedBioFact}:{" "}
        </StatBioTypography>
        {selectedHero.biography[bioFact]}
      </ListItemText>
    );
  });

  const statsList = stats.map((stat) => {
    const capitalizedStat = stat[0].toUpperCase() + stat.slice(1);
    return (
      <ListItemText>
        <StatBioTypography variant={"p"}>{capitalizedStat}: </StatBioTypography>
        {selectedHero.powerstats[stat]}
      </ListItemText>
    );
  });

  return (
    <Card>
      <HeaderWrapper>
        <CardHeader title={selectedHero.name} />
        <IconButton onClick={closeSelectedHero}>
          <CloseIcon />
        </IconButton>
      </HeaderWrapper>
      <CardContentWrapper>
        <HeroCardMedia
          image={selectedHero.image.url}
          title={selectedHero.name}
        />
        <CardTextWrapper>
          <List>{bioFactsList}</List>
          <Divider />
          <List>{statsList}</List>
        </CardTextWrapper>
      </CardContentWrapper>
    </Card>
  );
};

export default HeroCard;
