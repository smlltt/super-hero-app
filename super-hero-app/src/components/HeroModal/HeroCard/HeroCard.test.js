import React from "react";
import renderer from "react-test-renderer";
import HeroCard from "./index";

it("renders with selected hero", () => {
  const tree = renderer
    .create(
      <HeroCard
        selectedHero={{
          id: 1,
          biography: { aliases: "none" },
          powerstats: { intelligence: "none" },
          image: { url: "none" },
        }}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
