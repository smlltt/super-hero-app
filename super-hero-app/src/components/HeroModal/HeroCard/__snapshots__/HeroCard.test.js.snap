// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with selected hero 1`] = `
<div
  className="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    className="sc-eCApnc cXfJFo"
  >
    <div
      className="MuiCardHeader-root"
    >
      <div
        className="MuiCardHeader-content"
      />
    </div>
    <button
      className="MuiButtonBase-root MuiIconButton-root"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiIconButton-label"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
          />
        </svg>
      </span>
    </button>
  </div>
  <div
    className="sc-gtsrHT gzzItF"
  >
    <div
      className="MuiCardMedia-root sc-dlnjwi fAivH"
      style={
        Object {
          "backgroundImage": "url(\\"none\\")",
        }
      }
    />
    <div
      className="sc-hKFxyN gFOcZr"
    >
      <ul
        className="MuiList-root MuiList-padding"
      >
        <div
          className="MuiListItemText-root"
        >
          <span
            className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
          >
            <span
              className="MuiTypography-root sc-jSFjdj ezAAwN"
            >
              Full name
              :
               
            </span>
          </span>
        </div>
        <div
          className="MuiListItemText-root"
        >
          <span
            className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
          >
            <span
              className="MuiTypography-root sc-jSFjdj ezAAwN"
            >
              Alter egos
              :
               
            </span>
          </span>
        </div>
        <div
          className="MuiListItemText-root"
        >
          <span
            className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
          >
            <span
              className="MuiTypography-root sc-jSFjdj ezAAwN"
            >
              Aliases
              :
               
            </span>
            none
          </span>
        </div>
        <div
          className="MuiListItemText-root"
        >
          <span
            className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
          >
            <span
              className="MuiTypography-root sc-jSFjdj ezAAwN"
            >
              Place of birth
              :
               
            </span>
          </span>
        </div>
        <div
          className="MuiListItemText-root"
        >
          <span
            className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
          >
            <span
              className="MuiTypography-root sc-jSFjdj ezAAwN"
            >
              First appearance
              :
               
            </span>
          </span>
        </div>
        <div
          className="MuiListItemText-root"
        >
          <span
            className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
          >
            <span
              className="MuiTypography-root sc-jSFjdj ezAAwN"
            >
              Publisher
              :
               
            </span>
          </span>
        </div>
        <div
          className="MuiListItemText-root"
        >
          <span
            className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
          >
            <span
              className="MuiTypography-root sc-jSFjdj ezAAwN"
            >
              Alignment
              :
               
            </span>
          </span>
        </div>
      </ul>
      <hr
        className="MuiDivider-root"
      />
      <ul
        className="MuiList-root MuiList-padding"
      >
        <div
          className="MuiListItemText-root"
        >
          <span
            className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
          >
            <span
              className="MuiTypography-root sc-jSFjdj ezAAwN"
            >
              Intelligence
              : 
            </span>
            none
          </span>
        </div>
        <div
          className="MuiListItemText-root"
        >
          <span
            className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
          >
            <span
              className="MuiTypography-root sc-jSFjdj ezAAwN"
            >
              Strength
              : 
            </span>
          </span>
        </div>
        <div
          className="MuiListItemText-root"
        >
          <span
            className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
          >
            <span
              className="MuiTypography-root sc-jSFjdj ezAAwN"
            >
              Speed
              : 
            </span>
          </span>
        </div>
        <div
          className="MuiListItemText-root"
        >
          <span
            className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
          >
            <span
              className="MuiTypography-root sc-jSFjdj ezAAwN"
            >
              Durability
              : 
            </span>
          </span>
        </div>
        <div
          className="MuiListItemText-root"
        >
          <span
            className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
          >
            <span
              className="MuiTypography-root sc-jSFjdj ezAAwN"
            >
              Power
              : 
            </span>
          </span>
        </div>
        <div
          className="MuiListItemText-root"
        >
          <span
            className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
          >
            <span
              className="MuiTypography-root sc-jSFjdj ezAAwN"
            >
              Combat
              : 
            </span>
          </span>
        </div>
      </ul>
    </div>
  </div>
</div>
`;
